cookiecutter/generate.py

```diff
@@ -72,18 +72,19 @@ def apply_overwrites_to_context(context, overwrite_context):
             context[variable] = overwrite
 
 
-def generate_context(context_file='cookiecutter.json', default_context=None,
-                     extra_context=None):
+def generate_context(repo_dir, context_file='cookiecutter.json',
```

Comment:
Why did you add the parameter in the beginning? That could potentially have breaking changes. It's safer if you move it at the end